crosshair/run_functional_tests.py

```diff
@@ -0,0 +1,252 @@
+                            continue
+                        else:
+                            return -1
+
+                elif outcome.startswith("bugs_detected"):
+                    if proc.returncode == 0:
+                        print(
+                            f"The functional test failed on an example. "
+                            f"Expected a failure (a return code not equal 0), "
+                            f"but got a return code {proc.returncode}.\n\n"
+                            f"The stdout was:\n{stdout}\n\n"
+                            f"The stderr was:\n{stderr}\n\n"
+                            f"The command was:\n{cmd_as_string}",
+                            file=sys.stderr)
+
+                        if continue_on_error:
+                            success = False
+                            continue
+                        else:
+                            return -1
+                else:
+                    raise AssertionError(f"Unexpected outcome: {outcome!r}")
+
+                expected_stdout_pth = (
+                        pth.parent / (pth.stem + ".out")
+                )
+
+                expected_stderr_pth = (
+                        pth.parent / (pth.stem + ".err")
+                )
+
+                ##
+                # Replace the absolute path to the examples directory
+                # with a place holder to make these tests machine agnostic.
+                ##
+
+                stdout = stdout.replace(str(examples_dir), "<examples dir>")
+                stderr = stderr.replace(str(examples_dir), "<examples dir>")
+
+                if overwrite:
+                    expected_stdout_pth.write_text(stdout)
+                    expected_stderr_pth.write_text(stderr)
+                else:
+                    if not expected_stdout_pth.exists():
+                        print(
+                            f"The golden stdout file does not exist: "
+                            f"{strip_cwd(expected_stdout_pth)}. "
+                            f"Invoke {strip_cwd(this_path)} with --overwrite?",
+                            file=sys.stderr)
+                        if continue_on_error:
+                            success = False
+                            continue
+                        else:
+                            return -1
+
+                    if not expected_stderr_pth.exists():
+                        print(
+                            f"The golden stderr file does not exist: "
+                            f"{strip_cwd(expected_stdout_pth)}. "
+                            f"Invoke {strip_cwd(this_path)} with --overwrite?",
+                            file=sys.stderr)
+                        if continue_on_error:
+                            success = False
+                            continue
+                        else:
+                            return -1
+
+                    expected_stdout = expected_stdout_pth.read_text()
```

Comment:
Can we normalize the path slashes in the output?
Or cut off the leading directories, e.g.?:
```
expected = re.compile(r'^.*[/\\]([_\w]+\.py)').sub(r'\1', expected)
```

Reply:
Merging so you can get that formatting PR through and we can iterate on this separately. (I'll want to tweak some of the examples myself too)

Reply:
> Can we normalize the path slashes in the output?

Silly me, thanks for spotting it! `.as_posix()` does the trick. Let me create a small PR to fix this.

Reply:
Sorry, `as_posix` was a dummy idea. I implemented your suggestion with regular expression, see https://github.com/pschanely/CrossHair/pull/64.